roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -0,0 +1,380 @@
+use async_channel::{Receiver, Sender};
+use codec_sv2::{
+    framing_sv2::framing::Frame, HandshakeRole, Initiator, Responder, StandardEitherFrame,
+};
+use key_utils::{Secp256k1PublicKey, Secp256k1SecretKey};
+use network_helpers_sv2::noise_connection_tokio::Connection;
+use roles_logic_sv2::{
+    parsers::{
+        CommonMessages,
+        JobDeclaration::{
+            AllocateMiningJobToken, AllocateMiningJobTokenSuccess, DeclareMiningJob,
+            DeclareMiningJobError, DeclareMiningJobSuccess, IdentifyTransactions,
+            IdentifyTransactionsSuccess, ProvideMissingTransactions,
+            ProvideMissingTransactionsSuccess, SubmitSolution,
+        },
+        PoolMessages,
```

Comment:
Please use `AnyMessage<'a>` in `protocols/v2/roles-logic-sv2/src/parsers.rs` instead of `PoolMessages`